src/main/java/mods/railcraft/common/worldgen/OreGeneratorFactory.java

```diff
@@ -213,7 +222,7 @@ public boolean isRichBiome(Biome biome) {
                 return level == WhiteBlackList.PermissionLevel.WHITELISTED;
 
             // FIXME this is wrong, it doesn't have to match all
-            return BiomeDictionary.getTypes(biome).stream().allMatch(richBiomeTypes::permits);
+            return BiomeDictionary.getTypes(biome).stream().anyMatch(richBiomeTypes::permits);
```

Comment:
This whole thing is no good. If any type is blacklisted, it should not be allowed. This check only checks that there is a type that is allowed which is also not blacklisted.

For example, you could blacklist Forest, and whitelist Mountains. The documentation indicates that a Biome of type "Mountainous Forests" would be blacklisted because it has Forest as a type. But the current implementation allows it because it has Mountain as a type and mountain isn't Blacklisted, ignoring the fact that Forest is blacklisted.

A proper solution must first check if the biome list intersects the blacklist, and if not, then check if the biome list intersects the whitelist.

Reply:
Fixed.